src/auth.ts

```diff
@@ -3,60 +3,73 @@ import * as os from 'os';
 import * as path from 'path';
 import * as core from '@actions/core';
 import * as io from '@actions/io';
+import {create as xmlCreate} from 'xmlbuilder2';
+import * as constants from './constants';
 
 export const M2_DIR = '.m2';
 export const SETTINGS_FILE = 'settings.xml';
 
-export const DEFAULT_ID = 'github';
-export const DEFAULT_USERNAME = 'GITHUB_ACTOR';
-export const DEFAULT_PASSWORD = 'GITHUB_TOKEN';
-
 export async function configAuthentication(
-  id = DEFAULT_ID,
-  username = DEFAULT_USERNAME,
-  password = DEFAULT_PASSWORD
+  id: string,
+  username: string,
+  password: string,
+  gpgPassphrase: string | undefined = undefined
 ) {
   console.log(
     `creating ${SETTINGS_FILE} with server-id: ${id};`,
-    `environment variables: username=\$${username} and password=\$${password}`
+    'environment variables:',
+    `username=\$${username},`,
+    `password=\$${password},`,
+    `and gpg-passphrase=${gpgPassphrase ? '$' + gpgPassphrase : null}`
```

Comment:
We should not log these, should we?

Reply:
This is actually fine, see the output: https://github.com/jaredpetersen/setup-java-gpg/runs/876184775?check_suite_focus=true#step:3:18

In `settings.xml` you can have interpolation: https://maven.apache.org/settings.html#quick-overview with things such as `${env.HOME}` so that nothing gets actually printed to the file. That's what we're doing, just with our own env variables such as $GITHUB_TOKEN

Reply:
If all we are printing out are variable names, then we really don't need to print this right?  It also creates a mental burden for others to check whether we are printing sensitive information or not. 😄 

My vote is to never have anything remotely resembles `print(password)` anywhere.  😃   We could print `password is not null` if that info is needed.

Since this isn't a new problem, this does not block the PR.

Reply:
I think this might help to confirm for example if passphrase is set or not, with `${gpgPassphrase ? '$' + gpgPassphrase : null}` But agreed on the other just `print(pwd)` kinds :)

Reply:
So just to be clear -- we're OK with this current output?